### PR TITLE
Fix non-json outputs

### PR DESCRIPTION
**Pull request by Coriou**
Sometimes youtube-dl won't return JSON. It can return a simple string (when using the "-g" or "--get-url" flags for instance). This commit, with the minimum amount of modifications possible, tries to handle this case.

Comes with a test

I think it's be cool to know what other flags can potentially output non-json results to test them all.

Discussed in #198

### CODE REVIEW

File: lib/youtube-dl.js [hunk 1]
```diff
@@ -238,6 +238,10 @@ ytdl.exec = function exec(url, args, options, callback) {
  * @returns {Object}
  */
 function parseInfo(data) {
+  // Youtube-dl might return just an url as a string when using the "-g" or "--get-url" flag
+  if (typeof data === "string" && data.startsWith("http"))
```
Comment (Kikobeats) [inline lib/youtube-dl.js]: not a requisite but appreciated if you can extract this into a function

```js
const isString = str => typeof str === "string"
```

and then use the function there and the rest of the places, like [here](https://github.com/przemyslawpluta/node-youtube-dl/blob/113ed2a1d61e434f35d7fa173770043c80704ddd/lib/youtube-dl.js#L191).

File: lib/youtube-dl.js [hunk 2]
```diff
@@ -238,6 +238,10 @@ ytdl.exec = function exec(url, args, options, callback) {
  * @returns {Object}
  */
 function parseInfo(data) {
+  // Youtube-dl might return just an url as a string when using the "-g" or "--get-url" flag
```
Comment (Kikobeats) [inline lib/youtube-dl.js]: Little typo: `Youtube-dl` → `youtube-dl`

File: lib/youtube-dl.js [hunk 3]
```diff
@@ -316,6 +320,12 @@ ytdl.getInfo = function getInfo(url, args, options, callback) {
     if (err) return callback(err);
     let info;
 
+    // If using the "-g" or "--get-url" flag youtube-dl will return just a string (the URL to the video) which messes up the parsing
+    // This fixes this behaviour
+    if (args && (args.indexOf("-g") > -1 || args.indexOf("--get-url") > -1))
```
Comment (Kikobeats) [inline lib/youtube-dl.js]: The same thing here, this kind of checks are around the code, could be nice start extracting it to make the code more readable:

```js
const has = (collection, value) => collection && collectionindexOf(value)  > -1
```

File: lib/youtube-dl.js (hunk 1 above)
Comment (Coriou) [reply lib/youtube-dl.js]: Done in e3f4a9d

File: lib/youtube-dl.js (hunk 2 above)
Comment (Coriou) [reply lib/youtube-dl.js]: Done in e968072

File: lib/youtube-dl.js (hunk 3 above)
Comment (Coriou) [reply lib/youtube-dl.js]: Done in 5ec1f87

Left [this](https://github.com/przemyslawpluta/node-youtube-dl/blob/113ed2a1d61e434f35d7fa173770043c80704ddd/lib/youtube-dl.js#L313) and [this](https://github.com/przemyslawpluta/node-youtube-dl/blob/113ed2a1d61e434f35d7fa173770043c80704ddd/lib/youtube-dl.js#L376) untouched as they're checking for a special index didn't want to break anything that's beyond the scope of this PR already.

**Comment by Coriou** [comment]
> LGTM; The only action to perform in order to merge this removes `package-lock.json` since we are not interested in maintaining it.

Removed in 56cbc63, my bad

> I added some comments most as recommendations, feel free to address them, very appreciated if you do that.

Addressed each of them in different commits, feel free to hand pick them

**Comment by Coriou** [comment]
Not sure what's causing the last build to change, seems like Youtube isn't returning the expected format. Passing in Node 8-11 on my local machine

**Comment by Kikobeats** [comment]
tests are not deterministic; 

That's a legacy thing and sometimes the video format can be different (webp, mp4).

**Comment by Kikobeats** [comment]
Green! 🎉

**Comment by Coriou** [comment]
Thank you for looking into this. Also fixes #198

**Pull request merged**